src/main/java/io/confluent/copycat/hdfs/TopicPartitionWriter.java

```diff
@@ -0,0 +1,461 @@
+/**
+ * Copyright 2015 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ **/
+
+package io.confluent.copycat.hdfs;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.copycat.errors.CopycatException;
+import org.apache.kafka.copycat.errors.IllegalWorkerStateException;
+import org.apache.kafka.copycat.sink.SinkRecord;
+import org.apache.kafka.copycat.sink.SinkTaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import io.confluent.copycat.avro.AvroData;
+import io.confluent.copycat.hdfs.partitioner.Partitioner;
+
+public class TopicPartitionWriter {
+  private static final Logger log = LoggerFactory.getLogger(TopicPartitionWriter.class);
+  private WAL wal;
+  private Map<String, String> tempfiles;
```

Comment:
I'm a bit confused how this works with more than one log file. In the case of time based partitioning, should there ever be more than one tempfile at a time? From what I can tell in this file, the partitioner isn't involved in deciding when to rotate files, so I'm not sure how we can correctly ensure exactly once delivery. Won't multiple open files here potentially break the way the WAL works?

Also, if the Partitioner isn't involved in choosing when to rotate, couldn't we end up with pretty bad file sizes? E.g. if the partitioner splits the data across many files, won't the files end up being quite small.